Lib/pathlib.py

```diff
@@ -430,6 +446,41 @@ def _parts_normcase(self):
             self._parts_normcase_cached = self._str_normcase.split(self._flavour.sep)
             return self._parts_normcase_cached
 
+    @property
+    def _lines(self):
+        # Path with separators and newlines swapped, for pattern matching.
+        try:
+            return self._lines_cached
+        except AttributeError:
+            trans = _SWAP_SEP_AND_NEWLINE[self._flavour.sep]
+            self._lines_cached = str(self).translate(trans)
+            return self._lines_cached
+
+    @property
+    def _matcher(self):
+        try:
+            return self._matcher_cached
+        except AttributeError:
+            if not self.parts:
+                raise ValueError("empty pattern")
```

Comment:
This should use `raise from None`, to suppress exception chaining, I think -- otherwise the traceback might imply to a user that that there's a bug in `pathlib`

```suggestion
                raise ValueError("empty pattern") from None
```

Reply:
Good spot!